### PR TITLE
Add verbose logging option

### LINKED ISSUE #89: Verbose flag for the CLI
The CLI should have a `--verbose` flag that enables the `DEBUG` logging level.

### PR DESCRIPTION
**Pull request by rayyaw**
## Changes
- Add verbose logging option.
- Refactor valid command line arguments into its own function.
- 6.2.1 -> 6.2.2

## Testing
- Automated testing.
- Added a debug log and verified it was printed with the `--verbose` flag.
- Added a debug log and verified it was not printed when the `--verbose` flag was not present.
- Made sure the valid command line arguments allowed the program to run.
- Made sure an invalid set of command line arguments caused the program to early exit.

## Related issues
- closes https://github.com/rayyaw/flash-patcher/issues/89

### CODE REVIEW

File: flash_patcher/patcher.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 from importlib.metadata import PackageNotFoundError, version
+# pylint: disable=no-name-in-module
```
Comment (GTcreyon) [inline flash_patcher/patcher.py:2]: Why is this needed?

**Review by GTcreyon — APPROVED** [review]
See comment. Also, it's a little hard to test if this works because I don't think there are any debug messages in the project.

File: flash_patcher/patcher.py (hunk 1 above)
Comment (rayyaw) [reply flash_patcher/patcher.py:2]: Pylint complains about `from logging import DEBUG`. I'm not entirely sure why.

I also tried this:
```
import logging
logger.setLevel(logging.DEBUG)
```

but here, pylint complains about `E1101: Module 'logging' has no 'DEBUG' member (no-member)`

I think we should keep the import as is since all of our other imports use `from` statements

**Comment by rayyaw** [comment]
> See comment. Also, it's a little hard to test if this works because I don't think there are any debug messages in the project.

I added my own debug logs for testing, you can add a `logger.debug()` in `patcher.py`
